inst/schema/report_run_request.json

```diff
@@ -0,0 +1,27 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "branch": {
+            "type": "string"
+        },
+        "hash": {
+            "type": "string"
+        },
```

Comment:
why do we have both branch and hash, and not ref?  Probably this is a good idea though tbh

What is the interpretation of branch names? Do we add main/ to them?

Reply:
the runner checks out to that branch and resets head to the hash, we would have a different behaviour if we used a ref right? would we not get a detached head if someone specified just a hash

there will be some extra work in the UI if we want people to be able to specify a commit hash on a branch instead of just a commit hash but that can come later

branch names are just what we do a git checkout to before resetting to the specified hash, im not sure what you mean by adding `main/` to them but we dont do anything like that, just pass branchname into git checkout via gert